docs/site/Controller-generator.md

```diff
@@ -106,43 +114,96 @@ export class TodoController {
     @repository(TodoRepository) public todoRepository: TodoRepository,
   ) {}
 
-  @post('/todos')
-  async create(@requestBody() obj: Todo): Promise<Todo> {
-    return await this.todoRepository.create(obj);
+  @post('/todos', {
+    responses: {
+      '200': {
+        description: 'Todo model instance',
+        content: {'application/json': {'x-ts-type': Todo}},
+      },
+    },
+  })
+  async create(@requestBody() data: Todo): Promise<Todo> {
+    return await this.todoRepository.create(data);
   }
 
-  @get('/todos/count')
+  @get('/todos/count', {
+    responses: {
+      '200': {
+        description: 'Todo model count',
+        content: {'application/json': {'x-ts-type': Number}},
+      },
+    },
+  })
   async count(@param.query.string('where') where?: Where): Promise<number> {
     return await this.todoRepository.count(where);
   }
 
-  @get('/todos')
-  async find(@param.query.string('filter') filter?: Filter): Promise<Todo[]> {
+  @get('/todos', {
+    responses: {
+      '200': {
+        description: 'Array of Todo model instances',
+        content: {
+          'application/json': {
+            schema: {type: 'array', items: {'x-ts-type': Todo}},
+          },
+        },
+      },
+    },
+  })
+  async find(
+    @param.query.object('filter', getFilterSchemaFor(Todo)) filter?: Filter,
```

Comment:
Out of scope of this PR I'd say but what is your thought on simplifying the DX here by creating the following:

```
@param.query.filter(Todo) => @param.query.object('filter', getFilterSchemaFor(Todo))
@param.query.filter('filterObj', Todo) => @param.query.object('filterObj', getFilterSchemaFor(Todo))
```

I know it goes against the normal `type` flavours we provide for param decorators but I do think it provides a cleaner developer experience.

Reply:
Sounds good to me, I have created a follow-up issue here: https://github.com/strongloop/loopback-next/issues/1749